static/js/home.js

```diff
@@ -185,40 +189,57 @@ function getEvents(){
         'facetObj':JSON.stringify(facetObj)
     };
 
-    if(maptToPlot == "roadsegments") {
-        jQuery.post(
-            "/get_segs",
-            bounddic,
-            getEventCB,
-            'json');
-    }else{
-        jQuery.post(
-            "/get_accidents",
-            bounddic,
-            getEventCB,
-            'json');
-    }
+  if(maptToPlot == "roadsegments") {
+    jQuery.post(
+        "/get_segs",
+        bounddic,
+        getEventCB,
+        'json');
+  }else{
+    jQuery.post(
+        "/get_accidents",
+        bounddic,
+        getEventCB,
+        'json');
+  }
+}
+
+function clearVisual(){
+  if(visual_mode == undefined)  return;
+  if(visual_mode === 'markers') {
+    set_markers(onscreenMarker, null); 
+  } else if(visual_mode === 'segments') {
+    setSegments(roadsegments, null);
+  } else {
+    visual_mode.setMap(null); 
+  }
 }
 
 function getEventCB(data){
+  if(lastShape != undefined){
```

Comment:
This function should be splitted into three function. 
getAccidentHeatmapCB
getAccidentMarkerCB
getRoadSegmentsCB